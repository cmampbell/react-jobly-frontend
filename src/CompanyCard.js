import React from 'react';
import { Card, CardBody, CardTitle, CardText} from 'reactstrap';
import { Link } from 'react-router-dom'

const CompanyCard = ({ company }) => {
    return (
        <Link to={`/companies/${company.handle}`}>
        <Card className="text-start" color="info" outline style={{ width: '80vw' }}>
            <CardBody>
                <CardTitle tag="h5">
                    {company.name}
                </CardTitle>
                <CardText>
                    Desc: {company.description}
                </CardText>
                <CardText>
                    Size: {company.numEmployees} employees
                </CardText>
            </CardBody>
        </Card>
        </Link>
    )
}

export default CompanyCard;