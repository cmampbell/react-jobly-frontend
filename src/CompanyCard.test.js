import { render } from '@testing-library/react';
import CompanyCard from './CompanyCard';
import { MemoryRouter } from 'react-router-dom';

describe('Smoke and snapshot CompanyCard test', () => {
    it('should render without crashing', () => {
        render(<CompanyCard company={{ 
                            handle: 'test', 
                            name: 'Name', 
                            numEmployees: 45, 
                            description: 'Test company' }}/>,
                            {wrapper: MemoryRouter});
    });
    it('should match snapshot', ()=> {
        const { asFragment } = render(<CompanyCard company={{ 
                                        handle: 'test', 
                                        name: 'Name', 
                                        numEmployees: 45, 
                                        description: 'Test company' }}/>, 
                                        {wrapper: MemoryRouter})
        expect(asFragment()).toMatchSnapshot();
    })
})