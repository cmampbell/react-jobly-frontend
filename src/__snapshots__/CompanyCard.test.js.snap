// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyCard test should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="/companies/test"
  >
    <div
      class="text-start card border-info"
      style="width: 80vw;"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Name
        </h5>
        <p
          class="card-text"
        >
          Desc: Test company
        </p>
        <p
          class="card-text"
        >
          Size: 45 employees
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
